packages/nextjs/components/Header.tsx

```diff
@@ -20,14 +56,65 @@ const NavLink = ({ href, children }: { href: string; children: React.ReactNode }
   );
 };
 
+const NavLinks = ({ setIsOpen }: { setIsOpen: React.Dispatch<React.SetStateAction<boolean>> }) => {
+  return (
+    <>
+      {menuItems.map((item, index) => {
+        return (
+          <li key={index}>
+            <NavLink href={item.link} setIsOpen={setIsOpen}>
+              {item.icon}
+              {item.label}
+            </NavLink>
+          </li>
+        );
+      })}
+    </>
+  );
+};
+
+/* End of possible Navlinks Components */
+
 /**
  * Site header
  */
 export default function Header() {
+  const [isOpen, setIsOpen] = useState(Boolean);
+  const ref = useRef<HTMLInputElement>(null);
+
+  // Is this considered accessing the DOM? Is this okay?
+  useEffect(() => {
+    function handler(event: { target: any }) {
+      if (!ref.current?.contains(event.target)) {
+        setIsOpen(false);
+      }
+    }
+    window.addEventListener("click", handler);
+    return () => window.removeEventListener("click", handler);
+  }, []);
+
   return (
     <div className="navbar bg-base-100 min-h-0 flex-shrink-0">
       <div className="navbar-start">
-        <div className="flex items-center	gap-2 mx-4">
+        <div className="sm:hidden dropdown" ref={ref}>
+          <button
+            className="btn btn-ghost hover:bg-secondary"
+            onClick={() => {
+              setIsOpen(!isOpen);
```

Comment:
It would be nice if we do : 
```diff
+ setIsOpen(prevIsOpenValue => !prevIsOpenValue)
- setIsOpen(!isOpen)
```
since we are more sure we will be getting more up-to-date value, just a suggestion